### PR TITLE
Add travis check for uppercase RFC2119 keywords and href checker

### PR DESCRIPTION
**Pull request by duglin**
And fixed 2 spots in the spec we missed

Also - added an href checker too.

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: tools/verify-all.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# default arg is root of our source tree
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPODIR=$(dirname "${BASH_SOURCE}")/..
+
+rc=0
+
```
Comment (duglin) [inline tools/verify-all.sh]: FYI - will be adding an href checker soon - what's why this file/indirection exists.  Just need to fix a bug in it first.

**Comment by avade** [comment]
Slightly concerned about introducing untested bash code into the project, mostly for future maintainability. I trust your coding @duglin :) 

Love the idea of verifying keywords are consistent via Travis.

Interested to hear what others think.

**Comment by duglin** [comment]
LGTM

**Comment by duglin** [comment]
I do plan on putting this bash into a repo with tests around it - just need more time...

**Comment by pmorie** [comment]
LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by duglin** [comment]
I've added an href checker and fixed some bad hrefs in the spec.
We'll need to re-do the LGTMs

**Comment by duglin** [comment]
LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by duglin** [comment]
rebased - sadly, more LGMTs needed

LGTM

**Comment by angarg12** [comment]
Ouch, LGTM

**Comment by duglin** [comment]
rebased, again

**Comment by pmorie** [comment]
LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by duglin** [comment]
ping @shalako @vaikas-google @fmui for the final LGTM

**Comment by vaikas-google** [comment]
LGTM

**Pull request merged**